src/wcs/src/main/java/org/geoserver/wcs/WCSInfoImpl.java

```diff
@@ -30,6 +30,8 @@ public class WCSInfoImpl extends ServiceInfoImpl implements WCSInfo {
 
     long maxOutputMemory = -1;
 
+    int defaultDeflateCompressionLevel = WCSInfo.DEFAULT_DEFLATE_COMPRESSION_LEVEL;
```

Comment:
I'm afraid that might not work, due to the way XStream builds objects form XML (skipping the normal construction path). 

Better to:
* Make it an Integer (just the field)
* Deal with eventual nulls in the getter, defaulting to the desired compression level.
* Update hashcode/tostring as a consequence